examples/quickstart/factorial.cpp

```diff
@@ -26,8 +26,8 @@ HPX_PLAIN_ACTION(factorial, factorial_action);
 ///////////////////////////////////////////////////////////////////////////////
 std::uint64_t factorial(std::uint64_t n)
 {
-    if (0 >= n)
-        return 1;
+    if (n < 2)
+        return n;
```

Comment:
This isn't right if `n = 0`. It should still always return `1`.

Reply:
This should never be called with `n == 0`, I think. In any case, the old test was wrong, definitely.

Reply:
I also think we do the same for all the other fibonacci examples.

Reply:
`0! = 1` is perfectly valid. Not a very interesting input for a parallel `factorial` but it's definitely valid input. The zeroth fibonacci number is actually `0` so it makes sense there.

Reply:
> The zeroth fibonacci number is actually 0 so it makes sense there.

Ok, then `if (n < 2) return n;` is actually correct?

Reply:
> Ok, then `if (n < 2) return n;` is actually correct?

Well, yes, for fibonacci. This is factorial ;)

Reply:
Uhh, right. Not sure what I was thinking...